src/sonar_widget/SonarPlot.cc

```diff
@@ -290,19 +316,9 @@ void SonarPlot::generateScanningTransferTable(const base::samples::Sonar& sonar)
 
             // pixels in the sonar image
             else {
-                int idBeam = theta.rad / motorStep.rad;
-
-                // top image
-                if (j <= height() / 2) {
-                    if (idBeam < 0) {
-                        idBeam = (numSteps - 1) + idBeam;
-                    }
-                }
-
-                // bottom image
-                else
-                    idBeam = numSteps / 2 - idBeam;
-
+                double angle = atan2(-point.x(), -point.y());
+                int idBeam = (numSteps / 2) + angle / motorStep.rad;
```

Comment:
Same comment about how to compute the idBeam ... probably best to add M_PI to the angle.